### PR TITLE
[NON JS | EP FORM] Awards lists implementation

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
[NON JS | EP FORM] Awards lists implementation

### CODE REVIEW

File: app/controllers/form/awards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      else
+        el["title"] == params[:title]
+      end
+    end
+
+    result_awards = if result_awards.present?
+      result_awards
+    else
+      []
+    end
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: `end` at 75, 4 is not aligned with `if` at 71, 20

File: app/controllers/form/awards_controller.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    @form_answer.document.merge(
+      input_name.to_sym => result_awards.to_json
+    )
+  end
+
+  def new
+  end
+
+  def create
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: Assignment Branch Condition size for create is too high. [15.03/15]

File: app/controllers/form/awards_controller.rb [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        el["title"] == params[:title] &&
+        el["year"] == params[:year]
+      else
+        el["title"] == params[:title]
+      end
+    end
+
+    result_awards = if result_awards.present?
+      result_awards
+    else
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: Align `else` with `if`.

File: app/controllers/form/awards_controller.rb [hunk 4]
```diff
@@ -0,0 +1,126 @@
+class Form::AwardsController < Form::BaseController
+
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: Extra empty line detected at class body beginning.

File: app/controllers/form/awards_controller.rb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      if params[:holder].present?
+        el["title"] == params[:title] &&
+        el["year"] == params[:year]
+      else
+        el["title"] == params[:title]
+      end
+    end
+
+    result_awards = if result_awards.present?
+      result_awards
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: Use 2 (not -14) spaces for indentation.

File: app/value_objects/award.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                   presence: true, if: "self.holder?"
+
+  # Should be 100 words maximum
+  validates :details, length: {
+    maximum: 100,
+    tokenizer: -> (str) { str.split },
+    message: "is too long (maximum is 100 words)"
+  }
+
+  def initialize(holder=false, attrs={})
```
Comment (matthewford) [inline app/value_objects/award.rb]: Surrounding space missing in default value assignment.

File: app/value_objects/award.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  validates :title, :details, presence: true
+
+  validates :title, length: { maximum: 100 }
+  validates :year, length: { maximum: 4 },
+                   presence: true, if: "self.holder?"
+
+  # Should be 100 words maximum
+  validates :details, length: {
+    maximum: 100,
```
Comment (dreamfall) [inline app/value_objects/award.rb]: It should be 56 (details_words_max is 50 so (50 + 10%).to_i + 1).

Comment (RuslanHamidullin) [inline app/value_objects/award.rb]: Nice catch, yeah it should be 50 words. Gonna fix

File: app/controllers/form/awards_controller.rb [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+    @form_answer.document.merge(
+      input_name.to_sym => result_awards.to_json
+    )
+  end
+
+  def new
+  end
+
+  def create
```
Comment (matthewford) [inline app/controllers/form/awards_controller.rb]: Assignment Branch Condition size for create is too high. [16.03/15]

File: app/validators/question_words_validator.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
+  # Words counter (limit + 10%).to_i + 1)
```
Comment (matthewford) [inline app/validators/question_words_validator.rb]: Incorrect indentation detected (column 2 instead of 0).

File: app/value_objects/award.rb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+  validates :title, :details, presence: true
+
+  validates :title, length: { maximum: 100 }
+  validates :year, length: { maximum: 4 },
+                   presence: true, if: "self.holder?"
+
+  validates_with QuestionWordsValidator, field_name: :details
+
+  def initialize(question, holder=false, attrs={})
```
Comment (matthewford) [inline app/value_objects/award.rb]: Surrounding space missing in default value assignment.

**Pull request merged**